wormchain/x/wormhole/ante/ante.go

```diff
@@ -0,0 +1,68 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/wormhole-foundation/wormchain/x/wormhole/keeper"
+)
+
+// Reject all messages if we're expecting a software update.
+type WormholeAllowlistDecorator struct {
+	k keeper.Keeper
+}
+
+func NewWormholeAllowlistDecorator(k keeper.Keeper) WormholeAllowlistDecorator {
+	return WormholeAllowlistDecorator{
+		k: k,
+	}
+}
+
+func (wh WormholeAllowlistDecorator) AnteHandle(request sdk.Request, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Request, err error) {
+	if request.IsReCheckTx() {
+		return next(request, tx, simulate)
+	}
+	if request.BlockHeight() < 1 {
+		// Don't reject gen_tx transactions
+		return next(request, tx, simulate)
+	}
+
+	verified_addresses := make(map[string]bool)
+
+	// For every message we should check:
+	// 1. There is an allowlist entry for the signer(s), OR
+	// 2. The signer(s) are validators in current or future guardian set.
+	// We cache the result for performance, since this is a ante handler that runs for everything.
+	for _, msg := range tx.GetMsgs() {
+		for _, signer := range msg.GetSigners() {
```

Comment:
Shouldn't we be ok if one signer is allowlisted?

Reply:
actually you're correct, i was thinking messages were individually signed for some reason